javascript/modules/client/src/internal/client-core.ts

```diff
@@ -68,6 +77,28 @@ export abstract class ClientCore implements IClientCore {
     return this.signer.provider ? this.signer : this.signer.connect(this.web3);
   }
 
+  get maxFeePerGas(): Promise<BigNumber> {
+    return this.connectedSigner
+      .getFeeData()
+      .then(
+        feeData =>
+          feeData.maxFeePerGas ??
+          Promise.reject(new Error("Cannot estimate gas"))
+      );
+  }
+
+  get feeEstimationReducer(): BigNumber {
+    let estimationReducer = 100;
+    if (
+      this._gasFeeReducer &&
+      this._gasFeeReducer > 0 &&
+      this._gasFeeReducer < 1
+    ) {
+      estimationReducer = Math.trunc((1 / this._gasFeeReducer) * 100);
```

Comment:
Related to the message about dividing, this feels confusing to me.
I would expect things to be as simple as `fee = maxFee * avgFactor` and that's it. The double division feels counter intuitive.

Reply:
In addition, we should only allow to execute valid cases only.
Meaning:
- If `_gasFeeReducer` is not set, we return a factor of `1`.
- If `_gasFeeReducer` is out of `[0, 1]` then we throw an error
- Otherwise, we return a multiplicable value

Reply:
> I would expect things to be as simple as fee = maxFee * avgFactor and that's it. The double division feels counter intuitive.

It was also for me initially, the problem comes when dealing with `BigNumber` and to avoid losing accuracy by converting this to another primitive type. Therefore the only chance I got was converting the factor to `BigNumber` also, and that's why it seems confusing.



> In addition, we should only allow to execute valid cases only.
> Meaning:
> 
>     If _gasFeeReducer is not set, we return a factor of 1.
>     If _gasFeeReducer is out of [0, 1] then we throw an error
>     Otherwise, we return a multiplicable value

It's already being done, the only difference I see is the error throwing, which LGTM, because it means that an invalid value has been set. Right now it just converts it to `1`.